apps/wpcom-block-editor/src/wpcom/features/switch-to-classic-notice.js

```diff
@@ -0,0 +1,76 @@
+/* eslint-disable import/no-extraneous-dependencies */
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import domReady from '@wordpress/dom-ready';
+import ReactDOM from 'react-dom';
+import { useState, createInterpolateElement } from '@wordpress/element';
+import { withNotices, Notice } from '@wordpress/components';
+import url from 'url';
+/* eslint-enable import/no-extraneous-dependencies */
+
+const parsedEditorUrl = url.parse( window.location.href, true );
+
+const ClassicEditorNotice = () => {
+	const [ noticedDismissed, setNoticeDismissed ] = useState( false );
+
+	const dismissNotice = () => {
+		setNoticeDismissed( true );
+		localStorage.setItem( 'dismissedClassicEditorNotice', 'true' );
+	};
+
+	const noticeText = createInterpolateElement(
+		'If you prefer the Classic editor, you can always switch. Open the options menu by click in the button at the top right of your screen, and select <strong>Switch to Classic editor</strong>.',
+		{ strong: <strong /> }
+	);
+
+	const noticeContent = (
+		<div>
+			<h2>{ __( "You're using the most modern WordPress Editor." ) } </h2>
+			<p>{ noticeText }</p>
+		</div>
+	);
+
+	return (
+		! noticedDismissed && (
+			<Notice
+				className="edit-post-switch-to-classic-notice"
+				status="info"
+				isDismissible={ true }
+				onRemove={ dismissNotice }
+			>
+				{ noticeContent }
+			</Notice>
+		)
+	);
+};
+
+const NoticeComponent = withNotices( ClassicEditorNotice );
+
+const noticeDismissed = localStorage.getItem( 'dismissedClassicEditorNotice' );
+
+// Need to replace this check with one that checks for a specific parsedEditorUrl.query param
+// https://github.com/Automattic/wp-calypso/issues/41087.
+if ( parsedEditorUrl.query[ 'environment-id' ] === 'stage' && noticeDismissed !== 'true' ) {
+	domReady( () => {
+		const editInception = setInterval( () => {
```

Comment:
unfortunately this is the only way I can see to know when the editor is in a state to be able to inject a component into its dom structure

Reply:
Could using a MutationObserver here provide a benefit? Might not be worth the extra effort changing if this works and is good enough for the short life of this notice.

Reply:
I think we would still need the setTimeout to get the dom element to attach the mutationObserver to, but if this was a more permanent feature mutationObserver would be probably be useful to get around the issue of it moving location when the tabs were changed ... good suggestion though, it is easy to forget about things like mutationObserver when you get used to React doing hard work like this for you!

Reply:
sorry, was thinking about it from the wrong angle - attaching mutationObserver to the root node could work and remove the need for setTimeout .. would be an interesting experiment to see which of the two options was more performant ... something for a rainy day sometime :-)